### DISCUSSION TITLE
How to make a tree child element open another page

### DISCUSSION THREAD

**Discussion by Zakaria-ouertani** [zauberzeug/nicegui#623] [Q&A]
I can make the first parent open another with ```on_select``` but i didn't figure out how to make it's children do the same.
Sorry if this is already answered somewhere and I couldn't find

**Comment by falkoschindler** [comment]
Hi, @Zakaria-ouertani, please give a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) of what you already tried and explain what you expect to happen and what is happening instead. Thanks!

**Comment by Zakaria-ouertani** [comment]
My code is in this [repo](github.com/Zakaria-ouertani/PyAlgo_nicegui)
Example:
```python

    trees = expansion_trees()
    expansions_number = len(expansion_contents)
    expansion_width = 100 / expansions_number
    for i in range(0, expansions_number):
        with ui.column().style(
            "width:100%; border:solid black 1px; margin-top:3px;"
        ):
            with ui.expansion(f"{i + 1}. {expansion_contents[i][0]}", icon="book").classes("w-full h-full").style(
                    "font-size:20px; height:100%; padding:5px;"
            ):
                ui.tree(trees[0], label_key="id", on_select=lambda: ui.open("/Introduction")).props("no-transition accordion")


def expansion_trees():
    intro = [
        {
        "id": "L'algorithme",
        "children": 
            [
                {"id": "La Forme Générale 'un Algorithme"},
                {"id": "Les Syntaxes et Structures Slgorithmiques."}
            ]
        }
    ]
    return [intro]
```

The main page looks like this and clicking on "L'Algorithme" opens a page with ui.open()
![image](https://user-images.githubusercontent.com/75418838/227591041-395c5dd8-13d9-46d9-a4b2-083dfa7f78c9.png)
Now i need to make it's children to do the same (be it opening a different page or the same or a section)
![image](https://user-images.githubusercontent.com/75418838/227591910-1ca7d75a-f1eb-4f41-ad82-ed1c77622ce2.png)

**Reply by falkoschindler** [reply]
Sorry, but the code is not executable. The indentation is wrong and `expansion_contents` is undefined. Unfortunately, we don't have the resources to dive deep into other repositories (and even the link is broken).
Therefore it is crucial that you reduce the code to the bare minimum while keeping it self-contained and executable.

**Reply by Zakaria-ouertani** [reply]
> Sorry, but the code is not executable. The indentation is wrong and `expansion_contents` is undefined. Unfortunately, we don't have the resources to dive deep into other repositories (and even the link is broken). Therefore it is crucial that you reduce the code to the bare minimum while keeping it self-contained and executable.

Ok i'm working on that

**Reply by Zakaria-ouertani** [reply]
> Sorry, but the code is not executable. The indentation is wrong and `expansion_contents` is undefined. Unfortunately, we don't have the resources to dive deep into other repositories (and even the link is broken). Therefore it is crucial that you reduce the code to the bare minimum while keeping it self-contained and executable.

I just noticed that a child element does the same thing it's parent does
in this case opening the page
```python
from nicegui import ui
from cours import intro


def expansion_trees():
    intro = [
        {
        "id": "L'algorithme",
        "children": 
            [
                {"id": "La Forme Générale 'un Algorithme"},
                {"id": "Les Syntaxes et Structures Slgorithmiques."}
            ]
        }
    ]
    return [intro]


def show_tab():
    expansion_contents = [
            ["Introduction", intro.show],
        ]

    trees = expansion_trees()
    expansions_number = len(expansion_contents)
    expansion_width = 100 / expansions_number
    for i in range(0, expansions_number):
        with ui.column().style(
            "width:100%; border:solid black 1px; margin-top:3px;"
        ):
            with ui.expansion(f"{i + 1}. {expansion_contents[i][0]}", icon="book").classes("w-full h-full").style(
                    "font-size:20px; height:100%; padding:5px;"
            ):
                ui.tree(trees[0], label_key="id", on_select=lambda: ui.open("/Introduction")).props("no-transition accordion")


with ui.header(elevated=True).style('background-color: #282A35;'):
    # ui.button(on_click=lambda: hamburger_menu.toggle()).props('flat color=white icon=menu')
    with ui.tabs() as tabs:
        ui.tab('Cours', icon='book')


with ui.tab_panels(tabs, value='Cours').style("width:100%; height:100%;"):
    with ui.tab_panel('Cours'):
        show_tab()


@ui.page("/Introduction")
def show_i():
    with ui.header(elevated=True).style("background-color:#282A35;"):
        ui.button(on_click=lambda: ui.open("/")).props("flat color=white icon=home")


ui.run()
```

**Reply by falkoschindler** [reply]
As far as I can judge this code does exactly what you asked for (at least after fixing the import error, since I don't have `cours`).

**Reply by Zakaria-ouertani** [reply]
> As far as I can judge this code does exactly what you asked for (at least after fixing the import error, since I don't have `cours`).

Oh yeah i forgot to remove that ```import``` it's irrelevent in the minimal version i made


What i want it to do is for the child elements to open another page
Like this
![image](https://user-images.githubusercontent.com/75418838/227608412-8e9f6c41-729a-4eb2-8007-8ed6fbd255d3.png)

**Reply by falkoschindler** [reply]
Here is the example from https://nicegui.io/documentation#tree, where I only replaced `ui.notify` with `ui.open`. This basically opens a different page per node. Sure, this still needs some integration. But in principle it's as simple as this:

```py
ui.tree([
    {'id': 'numbers', 'children': [{'id': '1'}, {'id': '2'}]},
    {'id': 'letters', 'children': [{'id': 'A'}, {'id': 'B'}]},
], label_key='id', on_select=lambda e: ui.open(f'/{e.value}'))
```

**Reply by Zakaria-ouertani** [reply]
> Here is the example from https://nicegui.io/documentation#tree, where I only replaced `ui.notify` with `ui.open`. This basically opens a different page per node. Sure, this still needs some integration. But in principle it's as simple as this:
> 
> ```python
> ui.tree([
>     {'id': 'numbers', 'children': [{'id': '1'}, {'id': '2'}]},
>     {'id': 'letters', 'children': [{'id': 'A'}, {'id': 'B'}]},
> ], label_key='id', on_select=lambda e: ui.open(f'/{e.value}'))
> ```

Oh ok when i read the example i didn't get what the e since i wasn't famililar with lambda syntax 
I should've looked more into it
Thank you
